frontend/test/metabase/scenarios/dashboard-filters/reproductions/12720.cy.spec.js

```diff
@@ -0,0 +1,111 @@
+import { restore } from "__support__/e2e/cypress";
+import { SAMPLE_DATASET } from "__support__/e2e/cypress_sample_dataset";
+
+const { ORDERS } = SAMPLE_DATASET;
+
+// After January 1st, 2020
+const dashboardFilter = {
+  default: "2020-01-01~",
+  id: "d3b78b27",
+  name: "Date Filter",
+  slug: "date_filter",
+  type: "date/all-options",
+};
+
+const questionDetails = {
+  name: "12720_SQL",
+  native: {
+    query: "SELECT * FROM ORDERS WHERE {{filter}}",
+    "template-tags": {
+      filter: {
+        id: "1d006bb7-045f-6c57-e41b-2661a7648276",
+        name: "filter",
+        "display-name": "Filter",
+        type: "dimension",
+        dimension: ["field", ORDERS.CREATED_AT, null],
+        "widget-type": "date/month-year",
+        default: null,
+      },
+    },
+  },
+};
+
+describe.skip("issue 12720", () => {
+  beforeEach(() => {
+    cy.intercept("POST", "/api/card/1/query").as("cardQuery");
+
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should show QB question on a dashboard with filter connected to card without data-permission (metabase#12720)", () => {
+    // In this test we're using already present question ("Orders") and the dashboard with that question ("Orders in a dashboard")
+    cy.addFilterToDashboard({ filter: dashboardFilter, dashboard_id: 1 });
+
+    cy.createNativeQuestion(questionDetails).then(
+      ({ body: { id: SQL_ID } }) => {
+        cy.intercept("POST", `/api/card/${SQL_ID}/query`).as("sqlQuery");
+
+        cy.request("POST", "/api/dashboard/1/cards", {
+          cardId: SQL_ID,
+        }).then(({ body: { id: SQL_DASH_CARD_ID } }) => {
+          cy.log(
+            "Edit both cards (adjust their size and connect them to the filter)",
+          );
+
+          cy.request("PUT", "/api/dashboard/1/cards", {
+            cards: [
```

Comment:
I'd consider extracting the setting of this object to a function.

If I understood it correctly, we'd pass `dashboardFilter.id` as the argument and get back `{ cards: [……]}`

Same principle as setting the object consts above in this file.